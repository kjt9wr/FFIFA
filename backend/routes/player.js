const router = require('express').Router();
const Player = require('../models/player.model');

router.route('/').get((req, res) => {
  Player.find()
    .then(player => res.json(player))
    .catch(err => res.status(400).json('Unable to find player: ' + err));
});

router.route('/roster/:ownerID').get((req, res) => {
  Player.find({owner: req.params.ownerID })
    .then(player => res.json(player))
    .catch(err => res.status(400).json('Unable to find player: ' + err));
});

// add player to db
router.route('/add').post((req, res) => {
  const name = req.body.name;
  const price = Number(req.body.price);
  const keep = Boolean(req.body.keep);
  const position = req.body.position;
  const rank = Number(req.body.rank);
  const owner = ownersIDByName[req.body.ownerName];
  const sleeperId = req.body.sleeperId;
  const newPlayer = new Player({ name, price, keep, position, rank, owner, sleeperId });

  newPlayer.save()
  .then(() => res.json(name + ' added! ID: '+ newPlayer._id ))
  .catch(err => res.status(400).json('Unable to add player: ' + err));
});

router.route('/add/sleeperId').post((req, res) => {
  Player.findOne({ name: req.body.name })
    .then(player => {
      player.sleeperId = req.body.sleeperId;
      player.save()
          .then(() => res.json(player.name + ' sleeperId updated!'))
          .catch(err => res.status(400).json('Unable to update player: ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
}); 


// Updating a keeper
router.route('/update/:pid').post((req, res) => {
  Player.findById(req.params.pid)
    .then(player => {
      player.keep = req.body.keep;
      player.save()
          .then(() => res.json('Player updated!'))
          .catch(err => res.status(400).json('Unable to update player: ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
  });

  // Get all players for given position, ordered by rank
  router.route('/getAll/:pos').get((req, res) => {
    Player.find({position: req.params.pos })
    .then((data) => { 
      res.send({data: getAllPlayersOrderedByRank(data)}); 
    })
    .catch(err => res.status(400).json('Unable to find player: ' + err));
  })


// Change a player's power ranking
router.route('/rank').put((req, res) => {
  Player.findById(req.body.pid)
    .then(player => {
      player.rank = req.body.rank;
      player.save()
          .then(() => res.json(player.name + ' rank updated!'))
          .catch(err => res.status(400).json('Unable to save player ranking: ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
  });

  // Set a player's arbitration year
router.route('/arbitration').put((req, res) => {
  Player.findById(req.body.pid)
    .then(player => {
      player.arbYear = req.body.arbYear;
      player.save()
          .then(() => res.json(player.name + ' arbitration year updated!'))
          .catch(err => res.status(400).json('Unable to save player arbitration year: ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
  });

  // Change a player's keeper class
  router.route('/keeperClass').put((req, res) => {
    Player.findById(req.body.pid)
      .then(player => {
        player.keeperClass = req.body.keeperClass;
        player.save()
            .then(() => res.json(player.name + ' Keeper class updated!'))
            .catch(err => res.status(400).json('Unable to save keeper class : ' + err));
        })
        .catch(err => res.status(400).json('Unable to find player: ' + err));
  });

//reset all keeper classes
router.route('/reset/keeperClass').put((req, res) => {
Player.find().then(player => {
  player.keeperClass = 0; 
  player.save()
  .then(() => res.json(player.name + ' Keeper class updated!'))
  .catch(err => res.status(400).json('Unable to save keeper class : ' + err));
})

  });

// Change a player's supermax status
router.route('/superMax').put((req, res) => {
  Player.findById(req.body.pid)
    .then(player => {
      player.superMax.plan = req.body.superMaxPlan;
      player.superMax.year = req.body.superMaxYear;
      player.keep = true;
      player.save()
          .then(() => res.json(player.name + ' super max status updated!'))
          .catch(err => res.status(400).json('Unable to save player supermax : ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
  });

// Change a player's owner 
router.route('/update/owner').put((req, res) => {
  const ownerid = ownersIDByName[req.body.ownerName];
    Player.findById(req.body.pid)
    .then(player => {
      player.owner = ownerid;
      player.save()
          .then(() => res.json(player.name + ' owner updated!'))
          .catch(err => res.status(400).json('Unable to update player: ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
});

// Update roster
router.route('/update/roster/:ownerId').post((req, res) => {
  const playersToUpdate = req.body.players;
  Player.updateMany({
      sleeperId: {
        $in: playersToUpdate
      }}, {
        $set: { owner: req.params.ownerId }
      })
  .then(() => res.json('Updated!'))
});

// Updating a draft result
router.route('/draftResult').put((req, res) => {
  let ownerid = ownersIDByName[req.body.ownerName];
    Player.findById(req.body.pid)
    .then(player => {
      player.owner = ownerid;
      player.price = Math.trunc(req.body.price*1.2);
      player.save()
          .then(() => res.json(player.name + ' drafted by ' + req.body.ownerName +  ' at $' + req.body.price))
          .catch(err => res.status(400).json('Unable to update player: ' + err));
      })
      .catch(err => res.status(400).json('Unable to find player: ' + err));
  });

const getAllPlayersOrderedByRank = (playerList) => {
  return playerList.map((player) => {
    return {
      _id: player.id,
      name: player.name,
      rank: player.rank
   }
  }).sort((a,b) => a.rank - b.rank);
}

const ownersIDByName = {
  "Kevin" : "5e80d724b3bdaf3413316177",
  "Justin" : "5e80d930b3bdaf3413316189",
  "Alex" : "5e80dd6ab3bdaf34133161bd",
  "Luigi" : "5e80da66b3bdaf341331619b",
  "Christian" : "5e80e173b3bdaf3413316213",
  "Matt" : "5e80df96b3bdaf34133161ef",
  "Brent" : "5e80db62b3bdaf34133161ab",
  "Michael" : "5e80de37b3bdaf34133161cf",
  "Nikos" : "5e80dedcb3bdaf34133161dd",
  "Chinmay" : "5e80e07eb3bdaf3413316200",
  "Patrick" : "5e80e1dab3bdaf3413316225",
  "Jeff" : "5e80e1deb3bdaf3413316226",
};

module.exports = router;
