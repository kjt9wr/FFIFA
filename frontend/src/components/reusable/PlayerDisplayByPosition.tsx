import { Button, Col, Row, Table } from "reactstrap";
import { Player } from "../../interfaces/interfaces";
import { TRANSPARENT_TABLE_STYLE } from "../../utilities/constants";
import { POSITION } from "../../utilities/enumerations";

interface PlayerDisplayByPositionProps {
  isEditable: Boolean;
  playerList: Player[];
  removePlayerCallback?: (playerId: string) => void;
}

const filterKeepersByPosition = (keptPlayers: Player[], position: string) => {
  return keptPlayers
    ? keptPlayers.filter((player: Player) => player.position === position)
    : [];
};
/*
 * This component displays a table of players grouped by position
 */
const PlayerDisplayByPosition = (props: PlayerDisplayByPositionProps) => {
  const { removePlayerCallback, isEditable, playerList } = props;

  const keptQBs = filterKeepersByPosition(playerList, POSITION.QB);
  const keptRBs = filterKeepersByPosition(playerList, POSITION.RB);
  const keptWRs = filterKeepersByPosition(playerList, POSITION.WR);
  const keptTEs = filterKeepersByPosition(playerList, POSITION.TE);

  const renderPlayersForPosition = (position: string, players: Player[]) => (
    <Table borderless size="sm" hover responsive>
      <thead className="thead-light">
        <tr>
          <th>{position}</th>
        </tr>
      </thead>
      <tbody>
        {players.map((player: Player) => {
          return (
            <tr key={player.name}>
              {isEditable && removePlayerCallback ? (
                <td
                  style={TRANSPARENT_TABLE_STYLE}
                  onClick={() => {
                    removePlayerCallback(player.sleeperId);
                  }}
                >
                  <Button size="sm" outline className="mr-auto">
                    X
                  </Button>
                  {player.name} - ${player.price}
                </td>
              ) : (
                <td style={TRANSPARENT_TABLE_STYLE}>{player.name}</td>
              )}
            </tr>
          );
        })}
      </tbody>
    </Table>
  );

  return (
    <Row xs="2" md="4">
      <Col>{renderPlayersForPosition(POSITION.QB, keptQBs)} </Col>
      <Col>{renderPlayersForPosition(POSITION.RB, keptRBs)}</Col>
      <Col>{renderPlayersForPosition(POSITION.WR, keptWRs)}</Col>
      <Col>{renderPlayersForPosition(POSITION.TE, keptTEs)}</Col>
    </Row>
  );
};

export default PlayerDisplayByPosition;
