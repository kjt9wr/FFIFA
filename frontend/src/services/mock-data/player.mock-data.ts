import { Player } from "../../interfaces/interfaces";

export const MAXED_PLAYER: Player = {
  _id: "abc",
  keep: true,
  keeperClass: 3,
  name: "Najee Harris",
  owner: "",
  position: "RB",
  superMax: {
    plan: 5,
    signingYear: 2023,
  },
  price: 10,
  sleeperId: "abc",
};

export const ARBITRATED_TE: Player = {
  _id: "kittle",
  keep: true,
  keeperClass: 4,
  name: "George Kittle",
  owner: "",
  position: "TE",
  price: 50,
  sleeperId: "kittle",
};

// Kept Players

const KEPT_PLAYER = {
  keep: true,
  keeperClass: 1,
  owner: "",
};

export const MOCK_JAYDEN_DANIELS: Player = {
  ...KEPT_PLAYER,
  _id: "jayden_daniels",
  keeperClass: 2,
  name: "Jayden Daniels",
  position: "QB",
  price: 67,
  sleeperId: "jayden_daniels",
};

export const MOCK_LAMAR_JACKSON: Player = {
  ...KEPT_PLAYER,
  _id: "lamar_jackson",
  name: "Lamar Jackson",
  position: "QB",
  price: 37,
  sleeperId: "lamar_jackson",
};

export const MOCK_KYLER_MURRAY: Player = {
  ...KEPT_PLAYER,
  _id: "kyler_murray",
  name: "Kyler Murray",
  position: "QB",
  price: 37,
  sleeperId: "kyler_murray",
};

export const MOCK_PATRICK_MAHOMES: Player = {
  ...KEPT_PLAYER,
  _id: "patrick_mahomes",
  name: "Patrick Mahomes",
  position: "QB",
  price: 37,
  rank: 10,
  sleeperId: "patrick_mahomes",
};

export const MOCK_JALEN_HURTS: Player = {
  ...KEPT_PLAYER,
  _id: "jalen_hurts",
  name: "Jalen Hurts",
  position: "QB",
  price: 37,
  sleeperId: "jalen_hurts",
};

export const MOCK_BIJAN_ROBINSON: Player = {
  ...KEPT_PLAYER,
  _id: "bijan_robison",
  keeperClass: 2,
  name: "Bijan Robinson",
  position: "RB",
  price: 121,
  sleeperId: "bijan_robison",
};

export const MOCK_SAQUON_BARKLEY: Player = {
  ...KEPT_PLAYER,
  _id: "saquon_barkley",
  keeperClass: 2,
  name: "Saquon Barkley",
  position: "RB",
  price: 100,
  sleeperId: "saquon_barkley",
};

export const MOCK_JAHMYR_GIBBS: Player = {
  ...KEPT_PLAYER,
  _id: "jahmyr_gibbs",
  keeperClass: 2,
  name: "Jahmyr Gibbs",
  position: "RB",
  price: 98,
  sleeperId: "jahmyr_gibbs",
};

export const MOCK_JOSH_JACOBS: Player = {
  ...KEPT_PLAYER,
  ...KEPT_PLAYER,
  _id: "josh_jacobs",
  name: "Josh Jacobs",
  position: "RB",
  price: 81,
  rank: 12,
  sleeperId: "josh_jacobs",
};

export const MOCK_DERRICK_HENRY: Player = {
  ...KEPT_PLAYER,
  _id: "derrick_henry",
  name: "Derrick Henry",
  position: "RB",
  price: 79,
  firstKeepYear: 2023,
  rank: 6,
  sleeperId: "derrick_henry",
};

export const MOCK_NAJEE_HARRIS: Player = {
  ...KEPT_PLAYER,
  _id: "najee_harris",
  keeperClass: 3,
  name: "Najee Harris",
  sleeperId: "najee_harris",
  position: "RB",
  owner: "Patrick",
  superMax: {
    plan: 5,
    signingYear: 2022,
  },
  price: 130,
};

export const MOCK_BROCK_BOWERS: Player = {
  ...KEPT_PLAYER,
  _id: "brock_bowers",
  keeperClass: 2,
  name: "Brock Bowers",
  position: "TE",
  price: 37,
  sleeperId: "brock_bowers",
};

export const MOCK_SAM_LAPORTA: Player = {
  ...KEPT_PLAYER,
  _id: "sam_laporta",
  name: "Sam Laporta",
  position: "TE",
  price: 12,
  rank: 4,
  sleeperId: "sam_laporta",
};

export const MOCK_DRAKE_LONDON: Player = {
  ...KEPT_PLAYER,
  _id: "drake_london",
  name: "Drake London",
  position: "WR",
  price: 64,
  keeperClass: 2,
  sleeperId: "drake_london",
};

export const MOCK_AJ_BROWN: Player = {
  ...KEPT_PLAYER,
  _id: "aj_brown",
  name: "AJ Brown",
  position: "WR",
  price: 61,
  keeperClass: 2,
  sleeperId: "aj_brown",
};

export const MOCK_CEEDEE_LAMB: Player = {
  ...KEPT_PLAYER,
  _id: "ceedee_lamb",
  name: "CeeDee Lamb",
  position: "WR",
  price: 56,
  sleeperId: "ceedee_lamb",
};

export const MOCK_TYREEK_HILL: Player = {
  ...KEPT_PLAYER,
  _id: "tyreek_hill",
  name: "Tyreek Hill",
  position: "WR",
  price: 56,
  rank: 10,
  sleeperId: "tyreek_hill",
};

export const MOCK_JAMARR_CHASE: Player = {
  ...KEPT_PLAYER,
  _id: "jamarr_chase",
  name: "Jamarr Chase",
  position: "WR",
  price: 52,
  firstKeepYear: 2022,
  sleeperId: "jamarr_chase",
};

export const MOCK_MALIK_NABERS: Player = {
  ...KEPT_PLAYER,
  _id: "malik_nabers",
  name: "Malik Nabers",
  position: "WR",
  price: 51,
  sleeperId: "malik_nabers",
};

export const MOCK_JALEN_COKER: Player = {
  ...KEPT_PLAYER,
  _id: "jalen_coker",
  name: "Jalen Coker",
  position: "WR",
  price: 10,
  rank: 30,
  sleeperId: "jalen_coker",
};

export const MOCK_CHRIS_OLAVE: Player = {
  _id: "chris_olave",
  keep: false,
  keeperClass: 1,
  name: "Chris Olave",
  owner: "",
  position: "WR",
  price: 62,
  sleeperId: "chris_olave",
};

export const MOCKED_ROSTER = [
  MOCK_AJ_BROWN,
  MOCK_CEEDEE_LAMB,
  MOCK_JALEN_HURTS,
  MOCK_CHRIS_OLAVE,
];

export const MOCKED_ROSTER_WITH_SUPERMAX = [
  ...MOCKED_ROSTER,
  MOCK_NAJEE_HARRIS,
];
