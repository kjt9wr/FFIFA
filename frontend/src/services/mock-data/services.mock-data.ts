import { FreeAgentStorage, Player } from "../../interfaces/interfaces";
import {
  MAXED_PLAYER,
  MOCK_AJ_BROWN,
  MOCK_BIJAN_ROBINSON,
  MOCK_BROCK_BOWERS,
  MOCK_CEEDEE_LAMB,
  MOCK_DERRICK_HENRY,
  MOCK_DRAKE_LONDON,
  MOCK_JAHMYR_GIBBS,
  MOCK_JALEN_COKER,
  MOCK_JALEN_HURTS,
  MOCK_JAMARR_CHASE,
  MOCK_JAYDEN_DANIELS,
  MOCK_JOSH_JACOBS,
  MOCK_KYLER_MURRAY,
  MOCK_LAMAR_JACKSON,
  MOCK_MALIK_NABERS,
  MOCK_NAJEE_HARRIS,
  MOCK_PATRICK_MAHOMES,
  MOCK_SAM_LAPORTA,
  MOCK_SAQUON_BARKLEY,
  MOCK_TYREEK_HILL,
} from "./player.mock-data";

export const FRANCHISE_TAG_SAMPLE_DATA = {
  qbFranchisePrice: 25,
  keptQBs: [],
  rbFranchisePrice: 75,
  keptRBs: [MOCK_DERRICK_HENRY, MAXED_PLAYER],
  wrFranchisePrice: 50,
  keptWRs: [],
  teFranchisePrice: 30,
  keptTEs: [],
};

export const MOCKED_PENALTY_FEES = [
  { name: "Kevin", penaltyFee: 20 },
  { name: "Justin", penaltyFee: 50 },
  { name: "Matt", penaltyFee: 0 },
  { name: "Luigi", penaltyFee: 0 },
  { name: "Alex", penaltyFee: 30 },
];

export const KEPT_PLAYERS: Player[] = [
  MOCK_JAYDEN_DANIELS,
  MOCK_JALEN_HURTS,
  MOCK_KYLER_MURRAY,
  MOCK_LAMAR_JACKSON,
  MOCK_PATRICK_MAHOMES,
  MOCK_BIJAN_ROBINSON,
  MOCK_SAQUON_BARKLEY,
  MOCK_JAHMYR_GIBBS,
  MOCK_JOSH_JACOBS,
  MOCK_DERRICK_HENRY,
  MOCK_NAJEE_HARRIS,
  MOCK_BROCK_BOWERS,
  MOCK_SAM_LAPORTA,
  MOCK_AJ_BROWN,
  MOCK_DRAKE_LONDON,
  MOCK_CEEDEE_LAMB,
  MOCK_TYREEK_HILL,
  MOCK_JAMARR_CHASE,
  MOCK_MALIK_NABERS,
  MOCK_JALEN_COKER,
];

export const MOCK_FREE_AGENTS: FreeAgentStorage = {
  availableQBs: [MOCK_PATRICK_MAHOMES],
  availableRBs: [MOCK_DERRICK_HENRY, MOCK_JOSH_JACOBS],
  availableWRs: [MOCK_TYREEK_HILL, MOCK_JALEN_COKER],
  availableTEs: [MOCK_SAM_LAPORTA],
};
