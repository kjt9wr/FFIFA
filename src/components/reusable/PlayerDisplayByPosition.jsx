import React from 'react';

const displayPlayers = (playerList) => {
  return playerList.map(player => {
      return <tr key={player.name}><td> { player.name }</td></tr>
  })
}

const filterKeepersByPosition = (keptPlayers, position) => {
  return keptPlayers.filter(player =>Â player.position === position);
}

const renderPlayersForPosition = (position, players) => (
  <table className='table'>
    <thead className='thead-light'>
        <tr><th>{position}</th></tr>
    </thead>
    <tbody>{ displayPlayers(players) }</tbody>
  </table>
)

const playerDisplayByPosition = (props) => {
  const {playerList} = props;
  const keptQBs = filterKeepersByPosition(playerList, "QB");
  const keptRBs = filterKeepersByPosition(playerList, "RB");
  const keptWRs = filterKeepersByPosition(playerList, "WR");
  const keptTEs = filterKeepersByPosition(playerList, "TE");

  return (
    <div style={{ display: 'flex' }}>
      {renderPlayersForPosition("QB", keptQBs)}
      {renderPlayersForPosition("RB", keptRBs)}
      {renderPlayersForPosition("WR", keptWRs)}
      {renderPlayersForPosition("TE", keptTEs)}
    </div>
  )
}

export default playerDisplayByPosition;
