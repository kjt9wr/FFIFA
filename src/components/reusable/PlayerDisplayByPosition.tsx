import React from "react";
import { Table } from "reactstrap";
import { TiDelete } from "react-icons/ti";
import { Player } from "../../interfaces/interfaces";

interface PlayerDisplayByPositionProps {
  isEditable: Boolean;
  playerList: Player[];
  removePlayerCallback?: (playerId: string) => void;
}

const filterKeepersByPosition = (keptPlayers: Player[], position: string) => {
  return keptPlayers
    ? keptPlayers.filter((player: Player) => player.position === position)
    : [];
};
/*
 * This component displays a table of players grouped by position
 */
const PlayerDisplayByPosition = (props: PlayerDisplayByPositionProps) => {
  const { removePlayerCallback, isEditable, playerList } = props;

  const keptQBs = filterKeepersByPosition(playerList, "QB");
  const keptRBs = filterKeepersByPosition(playerList, "RB");
  const keptWRs = filterKeepersByPosition(playerList, "WR");
  const keptTEs = filterKeepersByPosition(playerList, "TE");

  const renderPlayersForPosition = (position: string, players: Player[]) => (
    <Table borderless size="sm" hover responsive>
      <thead className="thead-light">
        <tr>
          <th>{position}</th>
        </tr>
      </thead>
      <tbody>
        {players.map((player: Player) => {
          return (
            <tr key={player.name}>
              {isEditable ? (
                <td
                  onClick={() => {
                    removePlayerCallback(player._id);
                  }}
                >
                  <TiDelete />
                  {player.name} - ${player.price}
                </td>
              ) : (
                <td>{player.name}</td>
              )}
            </tr>
          );
        })}
      </tbody>
    </Table>
  );

  return (
    <div style={{ display: "flex" }}>
      {renderPlayersForPosition("QB", keptQBs)}
      {renderPlayersForPosition("RB", keptRBs)}
      {renderPlayersForPosition("WR", keptWRs)}
      {renderPlayersForPosition("TE", keptTEs)}
    </div>
  );
};

export default PlayerDisplayByPosition;
