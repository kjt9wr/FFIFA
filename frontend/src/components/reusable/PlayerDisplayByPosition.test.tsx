import { render, screen } from "@testing-library/react";
import PlayerDisplayByPosition from "./KeptPlayersDisplay";
import { MOCKED_ROSTER } from "../../services/mock-data/player.mock-data";
import userEvent from "@testing-library/user-event";

const removePlayer = jest.fn();
describe("player display by position", () => {
  it("renders static version successfully", () => {
    const view = render(
      <PlayerDisplayByPosition isEditable={false} playerList={MOCKED_ROSTER} />
    );
    expect(view).toMatchSnapshot();
  });

  it("calls removePlayerCallback on clicking the player", async () => {
    render(
      <PlayerDisplayByPosition
        isEditable={true}
        playerList={MOCKED_ROSTER}
        removePlayerCallback={removePlayer}
      />
    );

    userEvent.click(screen.getByText(/AJ Brown/));
    expect(removePlayer).toHaveBeenCalled();
  });
});
