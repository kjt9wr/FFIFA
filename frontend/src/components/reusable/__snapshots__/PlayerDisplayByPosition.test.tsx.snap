// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`player display by position renders static version successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row row-cols-2 row-cols-md-4"
      >
        <div
          class="col"
        >
          <div
            class="table-responsive"
          >
            <table
              class="table table-sm table-borderless table-hover"
            >
              <thead
                class="thead-light"
              >
                <tr>
                  <th>
                    QB
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="roster-display-table"
                >
                  <td>
                    Jalen Hurts
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
        </div>
        <div
          class="col"
        >
          <div
            class="table-responsive"
          >
            <table
              class="table table-sm table-borderless table-hover"
            >
              <thead
                class="thead-light"
              >
                <tr>
                  <th>
                    RB
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="table-responsive"
          >
            <table
              class="table table-sm table-borderless table-hover"
            >
              <thead
                class="thead-light"
              >
                <tr>
                  <th>
                    WR
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="roster-display-table"
                >
                  <td>
                    AJ Brown
                  </td>
                </tr>
                <tr
                  class="roster-display-table"
                >
                  <td>
                    CeeDee Lamb
                  </td>
                </tr>
                <tr
                  class="roster-display-table"
                >
                  <td>
                    Chris Olave
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="table-responsive"
          >
            <table
              class="table table-sm table-borderless table-hover"
            >
              <thead
                class="thead-light"
              >
                <tr>
                  <th>
                    TE
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row row-cols-2 row-cols-md-4"
    >
      <div
        class="col"
      >
        <div
          class="table-responsive"
        >
          <table
            class="table table-sm table-borderless table-hover"
          >
            <thead
              class="thead-light"
            >
              <tr>
                <th>
                  QB
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="roster-display-table"
              >
                <td>
                  Jalen Hurts
                </td>
              </tr>
            </tbody>
          </table>
        </div>
         
      </div>
      <div
        class="col"
      >
        <div
          class="table-responsive"
        >
          <table
            class="table table-sm table-borderless table-hover"
          >
            <thead
              class="thead-light"
            >
              <tr>
                <th>
                  RB
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="table-responsive"
        >
          <table
            class="table table-sm table-borderless table-hover"
          >
            <thead
              class="thead-light"
            >
              <tr>
                <th>
                  WR
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="roster-display-table"
              >
                <td>
                  AJ Brown
                </td>
              </tr>
              <tr
                class="roster-display-table"
              >
                <td>
                  CeeDee Lamb
                </td>
              </tr>
              <tr
                class="roster-display-table"
              >
                <td>
                  Chris Olave
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="table-responsive"
        >
          <table
            class="table table-sm table-borderless table-hover"
          >
            <thead
              class="thead-light"
            >
              <tr>
                <th>
                  TE
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
